### PR TITLE
Copy changes from 0.15.2 to master

### PR DESCRIPTION
**Pull request by radetsky**
I did cherry-picks from 0.15.2 to master. 

## Checklist

- [x] Change is covered by automated tests

### CODE REVIEW

File: docs/examples/android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.2-bin.zip
```
Comment (Lagovas) [inline docs/examples/android/gradle/wrapper/gradle-wrapper.properties:3]: can we use gradlew version for android the same as for java (8.0.2)?

File: docs/examples/objc/iOS-Carthage/Cartfile [hunk 2]
```diff
@@ -1 +1,2 @@
-github "cossacklabs/themis" ~> 0.14.0
\ No newline at end of file
+github "cossacklabs/themis" "0.15.1"
```
Comment (Lagovas) [inline docs/examples/objc/iOS-Carthage/Cartfile:1]: 0.15.1 or 0.15.2?

File: docs/examples/react-native/RNThemisExample/android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,123 @@
+apply plugin: "com.android.application"
+apply plugin: "com.facebook.react"
+
+/**
+ * This is the configuration block to customize your React Native Android app.
+ * By default you don't need to apply any configuration, just uncomment the lines you need.
+ */
+react {
+    /* Folders */
+    //   The root of your project, i.e. where "package.json" lives. Default is '..'
```
Comment (Lagovas) [inline docs/examples/react-native/RNThemisExample/android/app/build.gradle:10]: it's our comments or generated by some tool?

File: docs/examples/react-native/RNThemisExample/android/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.1-all.zip
```
Comment (Lagovas) [inline docs/examples/react-native/RNThemisExample/android/gradle/wrapper/gradle-wrapper.properties:3]: third version of gradlew

File: docs/examples/objc/iOS-Carthage/Cartfile (hunk 2 above)
Comment (radetsky) [reply docs/examples/objc/iOS-Carthage/Cartfile:1]: Apple-related wrappers have version 0.15.1
Would you recommend increasing all package versions to 0.15.2 without any changes?

File: docs/examples/react-native/RNThemisExample/android/app/build.gradle (hunk 3 above)
Comment (radetsky) [reply docs/examples/react-native/RNThemisExample/android/app/build.gradle:10]: It's generated.

**Pull request closed without merging**